features/support/form_helpers.rb

```diff
@@ -200,16 +212,184 @@ def fill_form(locator = nil, options = {})
     with = options.delete(:with) || {}
 
     values = {}
-
     maybe_within do
       find_fields.each do |name|
+        if find_fields(locator = name).length > 0
+          values[name] = (with[name] || random_for(name))
+          fill_field name, with: values[name]
+        end
+      end
+    end
 
-        values[name] = (with[name] || random_for(name))
+    values
+  end
+
+  def check_first_checkbox
+    checkbox = find_elements_by_xpath("//input[@type='checkbox']")[0]
+    checkbox.click
+  end
 
-        fill_field name, with: values[name]
+  def uncheck_all_checkboxes
+    checkboxes = find_elements_by_xpath("//input[@type='checkbox']")
+    checkboxes.each do |checkbox|
+      if checkbox.checked?
+        checkbox.click
       end
     end
+  end
 
-    values
+  def pass_text_validation
+    options = {}
+    questions = find_elements_by_xpath("//input[@class='text-box-with-error']")
+    if questions.length > 0
```

Comment:
May not need to check for `questions` length here. If it is 0 then nothing will happen

Reply:
cool, I will try without that if statement :)